boards/BMS/Src/Io/Io_LTC6813/Io_LTC6813CellVoltages.c

```diff
@@ -196,17 +196,12 @@ bool Io_LTC6813CellVoltages_ReadVoltages(void)
             Io_LTC6813Shared_PackCmdPec15(tx_cmd);
 
             // Transmit the command and receive data stored in register group
-            if (Io_SharedSpi_TransmitAndReceive(
+            if (!Io_SharedSpi_TransmitAndReceive(
                     ltc6813_spi, (uint8_t *)tx_cmd, TOTAL_NUM_CMD_BYTES,
-                    (uint8_t *)rx_buffer, TOTAL_NUM_OF_REG_BYTES))
+                    (uint8_t *)rx_buffer, TOTAL_NUM_OF_REG_BYTES) ||
+                !Io_ParseCellVoltageFromAllSegments(curr_reg_group, rx_buffer))
             {
-                // Parse cell voltages from the current register group from all
-                // segments
-                if (!Io_ParseCellVoltageFromAllSegments(
-                        curr_reg_group, rx_buffer))
-                {
-                    return false;
-                }
+                return false;
```

Comment:
Checking whether exiting the method here is intentional -- if you want to continue reading data from the remaining register groups after a failure this should be setting `status=false` instead.

Reply:
Actually i think my intention behind this was to discard the current snapshot of the voltages completely if a PEC15 mismatch occurs. Might not be necessary to do this though

Reply:
Yeah ok I figured that might be the case. Would be helpful to add a comment somewhere (here or header file) explaining how partial failures are handled

Reply:
should be nicer now @alxshev